src/modules/shopPage/components/FilterMenu/FilterFullMenu/FilterFullMenuDiametrData.tsx

```diff
@@ -0,0 +1,132 @@
+import { useState, useEffect } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { selectFilterData } from "../../../../mainPage/selectors";
+import { selectSearchInput, selectSelectedDiametr } from "../../../selectors";
+import { actions } from "../../../reducer";
+
+import ClearIcon from "@mui/icons-material/Clear";
+import { Box, Button, Typography } from "@mui/material";
+import { styled } from "@mui/material/styles";
+
+import { FILTER_COLORS, FONTS } from "../../../../../shared/constants";
+
+const ButtonsContainer = styled("div")({
+  display: "grid",
+  gridTemplateColumns: "repeat(3, 1fr)",
+  overflowY: "auto",
+  maxHeight: "472px",
+  width: "372px",
+});
+
+const StyledButton = styled(Button)({
+  height: "59px",
+  border: "none",
+  fontSize: "1.2rem",
+  fontFamily: FONTS.BOLD_TEXT_FAMILY,
+  color: `${FILTER_COLORS.TEXT_MAIN}`,
+  "&:hover": {
+    backgroundColor: `${FILTER_COLORS.BACKGROUND_GREY}`,
+    border: "none",
+    borderColor: "transparent",
+  },
+  "&:focus": {
+    borderColor: "transparent",
+  },
+});
+
+function FilterFullMenuDiametr() {
+  const dispatch = useDispatch();
+  const searchInput = useSelector(selectSearchInput);
+  const selectedDiametr = useSelector(selectSelectedDiametr);
+  const filtersParams = useSelector(selectFilterData());
+  const [filteredDiamenrOptions, setFilteredDiamenrOptions] = useState(
+    filtersParams.diametr
+  );
+
+  useEffect(() => {
+    const filtered = filtersParams.diametr.filter((option) =>
+      option.includes(searchInput)
+    );
+
+    setFilteredDiamenrOptions(filtered);
+  }, [searchInput, filtersParams.diametr]);
+
+  function handleDiametrClick(diametr: string) {
+    dispatch(actions.setSelectedDiametr(diametr));
+    dispatch(actions.toggleFullMenu());
+  }
+
+  function handleResetFilterDiametr() {
+    dispatch(actions.setSelectedDiametr(""));
+  }
+
+  return (
+    <>
+      <Box
+        onClick={handleResetFilterDiametr}
+        sx={{
+          display: "flex",
+          alignItems: "center",
```

Comment:
base `Box` styles

Reply:
Nope(
display: "block" is the default param